test/freenet/clients/http/utils/L10nExtensionTest.java

```diff
@@ -0,0 +1,63 @@
+package freenet.clients.http.utils;
+
+import java.util.Locale;
+import java.util.Map;
+
+import com.mitchellbosecke.pebble.extension.Function;
+import com.mitchellbosecke.pebble.template.EvaluationContext;
+import org.junit.Test;
+
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.singletonMap;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.nullValue;
+
+public class L10nExtensionTest {
+
+	@Test
+	public void l10nExtensionExposesAnL10nFunction() {
+		assertThat(l10nExtension.getFunctions().keySet(), contains("l10n"));
+	}
+
+	@Test
+	public void l10nFunctionDoesNotHaveArguments() {
+		assertThat(l10nFunction.getArgumentNames(), nullValue());
+	}
+
+	@Test
+	public void l10nFunctionReturnStringNullIfNotArgumentsAreGiven() {
+		assertThat(l10nFunction.execute(emptyMap(), null, null, 0), equalTo("null"));
+	}
+
+	@Test
+	public void l10nFunctionRetrievesGivenArgumentWithPrefixFromContextToNodeL10n() {
+		Map<String, Object> arguments = singletonMap("0", "l10nFunctionTest");
+		EvaluationContext context = createEvaluationContext();
+		assertThat(l10nFunction.execute(arguments, null, context, 0), equalTo("Localized Value"));
+	}
+
+	private static EvaluationContext createEvaluationContext() {
+		return new EvaluationContext() {
+			@Override
+			public boolean isStrictVariables() {
```

Comment:
should we enable strictVariables to have some error checking? (that does compile-time checks of the templates)

Reply:
I don’t think so. This context is only used in the test, and we don’t need to test that Pebble handles its contexts correctly. 🙂

Reply:
Ah, yes, that’s sensible. Thanks!